beacon-chain/core/transition/transition_no_verify_sig.go

```diff
@@ -296,14 +296,11 @@ func ProcessBlockForStateRoot(
 		}
 		if blk.IsBlinded() {
 			state, err = b.ProcessPayloadHeader(state, executionData)
-			if err != nil {
-				return nil, errors.Wrap(err, "could not process execution payload header")
-			}
 		} else {
 			state, err = b.ProcessPayload(state, executionData)
-			if err != nil {
-				return nil, errors.Wrap(err, "could not process execution payload")
-			}
+		}
+		if err != nil {
+			return nil, errors.Wrap(err, "could not process execution payload header")
```

Comment:
It's not always header now. Maybe: `could not process execution payload`

Reply:
fixed